include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -611,29 +616,31 @@ struct __result_and_scratch_storage
 #endif
     }
 
+    template <sycl::access_mode _AccessMode = sycl::access_mode::read_write>
     auto
-    __get_result_acc(sycl::handler& __cgh) const
+    __get_result_acc(sycl::handler& __cgh, const sycl::property_list& __prop_list = {}) const
     {
 #if _ONEDPL_SYCL_UNIFIED_USM_BUFFER_PRESENT
         if (__use_USM_host && __supports_USM_device)
-            return __usm_or_buffer_accessor<_T>(__cgh, __result_buf.get());
+            return __usm_or_buffer_accessor<_T, _AccessMode>(__cgh, __result_buf.get(), __prop_list);
         else if (__supports_USM_device)
-            return __usm_or_buffer_accessor<_T>(__cgh, __scratch_buf.get(), __scratch_n);
-        return __usm_or_buffer_accessor<_T>(__cgh, __sycl_buf.get(), __scratch_n);
+            return __usm_or_buffer_accessor<_T, _AccessMode>(__cgh, __scratch_buf.get(), __scratch_n, __prop_list);
+        return __usm_or_buffer_accessor<_T, _AccessMode>(__cgh, __sycl_buf.get(), __scratch_n, __prop_list);
 #else
-        return sycl::accessor(*__sycl_buf.get(), __cgh, sycl::read_write, __dpl_sycl::__no_init{});
+        return sycl::accessor(*__sycl_buf.get(), __cgh, _AccessMode, __prop_list);
```

Comment:
`_AccessMode` should be passed as a template parameter I believe. This path doesn't compile as there is no matching constructor.

The following should work:
```
 return sycl::accessor<_T, 1, _AccessMode>(*__sycl_buf.get(), __cgh, __prop_list);
```

Reply:
Alternatively, I think we could just switch to using the tags here.

Reply:
Done,
But is it enough?
Or we should write something like 
```C++
sycl::accessor<_T, 1, _AccessMode, __dpl_sycl::__target_device, sycl::access::placeholder::false_t>;
```
as it written at `struct __usm_or_buffer_accessor` ?

Reply:
Those additional parameters are defaulted to the same values that we explicitly list in `__usm_or_buffer_accessor` in the SYCL spec:
```
template <typename DataT, int Dimensions = 1,
          access_mode AccessMode =
              (std::is_const_v<DataT> ? access_mode::read
                                      : access_mode::read_write),
          target AccessTarget = target::device,
          access::placeholder isPlaceholder = access::placeholder::false_t>
class accessor { ... }
```
so the effect is the same. Although for consistency we should maybe use the same in both places.

Reply:
Yes I think we should use the whole set like the other places.

Reply:
And now we have only one definition of `__accessor_t` in the `struct __result_and_scratch_storage`:
```C++
    template <sycl::access_mode _AccessMode>
    using __accessor_t =
        sycl::accessor<_T, 1, _AccessMode, __dpl_sycl::__target_device, sycl::access::placeholder::false_t>;
```